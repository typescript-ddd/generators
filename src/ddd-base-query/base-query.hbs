import { Query } from "@typescript-ddd/core";

/**
 * Represents a base class for {{natural entityName}} queries.
 {{#each properties}}
 * @param {{lbrace}}{{type}}{{rbrace}} {{name}} - The {{natural name}} of the {{natural ../entityName}}.
  {{/each}}
 */
export abstract class {{pascal entityName}}Query implements Query {
  protected constructor(
    {{#each properties}}
    public readonly {{name}}: {{type}},
    {{/each}}
  ) {
    // TODO: Add validation here
  }
}
