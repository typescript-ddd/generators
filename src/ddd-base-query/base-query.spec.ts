import { BaseQueryTemplateData } from "./base-query.types";
import { generateBaseQuery } from "./base-query.generator";

describe("Query", () => {

  it("should render template", () => {
    // Arrange
    const data: BaseQueryTemplateData = {
      entityName: "User",
      properties: [
        { name: "id", type: "UserId" },
      ],
    };

    // Act
    const result = generateBaseQuery(data);

    // Assert
    expect(result).toMatchSnapshot();
  });
});
