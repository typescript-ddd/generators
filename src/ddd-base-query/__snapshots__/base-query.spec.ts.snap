// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query should render template 1`] = `
"import { Query } from "@typescript-ddd/core";

/**
 * Represents a base class for user queries.
 * @param {UserId} id - The id of the user.
 */
export abstract class UserQuery implements Query {
  protected constructor(
    public readonly id: UserId,
  ) {
    // TODO: Add validation here
  }
}
"
`;
